diambra/arena/arena_gym.py

```diff
@@ -2,95 +2,137 @@
+        self.print_actions_dict = [move_dict, attack_dict]
 
         # Maximum difference in players health
-        self.max_delta_health = env_info_dict["delta_health"]
+        for k in sorted(self.env_info.ram_states.keys()):
+            if "health" in k:
+                self.max_delta_health = self.env_info.ram_states[k].max - self.env_info.ram_states[k].min
+                break
+
+    def _get_ram_states_obs_dict(self):
+        player_spec_dict = {}
+        generic_dict = {}
+        # Adding env additional observations (side-specific)
+        for k, v in self.env_info.ram_states.items():
+            if k[-2:] == "P1":
+                target_dict = player_spec_dict
+                knew = "own_" + k[:-2]
+            elif k[-2:] == "P2":
+                target_dict = player_spec_dict
+                knew = "opp_" + k[:-2]
+            else:
+                target_dict = generic_dict
+                knew = k
+
+            # Discrete spaces (binary / categorical)
+            if v.type == 0 or v.type == 2:
+                target_dict[knew] = gym.spaces.Discrete(v.max + 1)
+            elif v.type == 1:  # Box spaces
+                target_dict[knew] = gym.spaces.Box(low=v.min, high=v.max, shape=(1,), dtype=np.int32)
+            else:
+                raise RuntimeError("Only Discrete (Binary/Categorical) | Box Spaces allowed")
+
+        player_spec_dict["action_move"] = gym.spaces.Discrete(self.n_actions[0])
+        player_spec_dict["action_attack"] = gym.spaces.Discrete(self.n_actions[1])
+
+        return generic_dict, player_spec_dict
+
+    # Get frame
+    def _get_frame(self, response):
+        self._frame = np.frombuffer(response.observation.frame, dtype='uint8').reshape(self.env_info.frame_shape.h, \
+                                                                                       self.env_info.frame_shape.w, \
+                                                                                       self.env_info.frame_shape.c)
+        return self._frame
 
-        # Maximum number of stages (1P game vs COM)
-        self.max_stage = env_info_dict["max_stage"]
+    # Get info
+    def _get_info(self, response):
+        info = dict(response.info.game_states)
+        info["settings"] = self.env_settings.pb_model
+        return info
 
-        # Min-Max reward
-        self.cumulative_reward_bounds = env_info_dict["cumulative_reward_bounds"]
+    # Integrate player specific RAM states into observation
+    def _player_specific_ram_states_integration(self, response, idx):
+        player_spec_dict = {}
+        generic_dict = {}
 
-        # Characters names list
-        self.char_names = env_info_dict["char_list"]
+        # Adding env additional observations (side-specific)
+        player_role = self.env_settings.pb_model.variable_env_settings.player_env_settings[idx].role
+        for k, v in self.env_info.ram_states.items():
+            if ("P1" in k or "P2" in k):
```

Comment:
That seems brittle, lets at least check if k endswith() Px

Reply:
For now I updated it using `endswith`, will keep this open to see if we will change the logic.